packages/fx-core/templates/core/v3Migration/js.ts.app.local.yml

```diff
@@ -17,6 +90,72 @@ deploy:
       {{/dotnet}}
 
   {{/tools}}
+  {{#tab}}
+  - uses: file/updateEnv
+    with:
+      target: ./tabs/.env.teamsfx.local
+      envs:
+        BROWSER: none
+        HTTPS: true
+        PORT: {{port}}
+        SSL_CRT_FILE: $\{{SSL_CRT_FILE}}
+        SSL_KEY_FILE: $\{{SSL_KEY_FILE}}
+
+  {{/tab}}
+  {{#if bot}}
+  - uses: file/updateEnv
+    with:
+      target: ./bot/.env.teamsfx.local
+      envs:
+        BOT_ID: $\{{BOT_ID}}
+        BOT_PASSWORD: $\{{SECRET_BOT_PASSWORD}}
+
+  {{/if}}
+  {{#ssoTab}}
+  - uses: file/updateEnv
+    with:
+      target: ./tabs/.env.teamsfx.local
+      envs:
+        REACT_APP_START_LOGIN_PAGE_URL: $\{{PROVISIONOUTPUT__AZURESTORAGETABOUTPUT__ENDPOINT}}/auth-start.html
+        REACT_APP_CLIENT_ID: $\{{AAD_APP_CLIENT_ID}}
+        {{#if functionName}}
+        REACT_APP_FUNC_ENDPOINT: http://localhost:7071
+        REACT_APP_FUNC_NAME: {{functionName}}
+        {{/if}}
+
+  {{/ssoTab}}
+  {{#if ssoBot}}
+  - uses: file/updateEnv
+    with:
+      target: ./bot/.env.teamsfx.local
+      envs:
+        M365_CLIENT_ID: $\{{AAD_APP_CLIENT_ID}}
+        M365_CLIENT_SECRET: $\{{SECRET_AAD_APP_CLIENT_SECRET}}
+        M365_TENANT_ID: $\{{AAD_APP_TENANT_ID}}
+        M365_AUTHORITY_HOST: $\{{AAD_APP_OAUTH_AUTHORITY_HOST}}
+        INITIATE_LOGIN_ENDPOINT: $\{{PROVISIONOUTPUT__AZUREWEBAPPBOTOUTPUT__SITEENDPOINT}}/auth-start.html
+        {{#if ssoTab}}
+        M365_APPLICATION_ID_URI: api://$\{{PROVISIONOUTPUT__AZURESTORAGETABOUTPUT__DOMAIN}}/botid-$\{{BOT_ID}}
```

Comment:
The `M365_APPLICATION_ID_URI` is different according project type. See `generateAppIdUri` in `packages\fx-core\src\core\middleware\utils\v3MigrationUtils.ts`

Reply:
Yes, so I use the if else syntax in this yml template.